astropy/coordinates/angles/core.py

```diff
@@ -180,9 +180,15 @@ def __new__(cls, angle, unit=None, dtype=np.inexact, copy=True, **kwargs):
                     # Possible conversion to `unit` will be done below.
                     angle = u.Quantity(angle, angle_unit, copy=COPY_IF_NEEDED)
 
-            elif isiterable(angle) and not (
-                isinstance(angle, np.ndarray) and angle.dtype.kind not in "SUVO"
-            ):
+            elif isinstance(angle, np.ndarray):
```

Comment:
I found the original logic here pretty hard to read and refactor to make room for a new check. This is what I got after a couple tries, which I think is clearer but also calls `isinstance` eagerly instead of lazily, which might incur a performance impact. I will run benchmarks on my next push.

Reply:
actually performance doesn't appear to be impacted (either in benchmarks or from simple measurement I did locally), so I'll remove the label now. For reference, here are logs from the latest run https://github.com/astropy/astropy/actions/runs/11557202273/job/32166662880